src/Microsoft.VisualStudio.SolutionPersistence/Model/SolutionArgumentException.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.VisualStudio.SolutionPersistence.Model;
+
+/// <summary>
+/// Reasons the SolutionArgumentException was raised.
+/// </summary>
+public enum SolutionArgumentExceptionType
```

Comment:
I think this could be renamed to SolutionError (or SolutionErrorCode) so it can also be used on SolutionException.